ImageChecks.ps1

```diff
@@ -0,0 +1,167 @@
+param (
+    [String]$Image,
+    [String]$ConfigFile
+)
+
+[String]$resourcesDir = "$pwd\UnattendResources"
+[String]$imgExtracted

```

Comment:
you do not need to cast variables or to declare them beforehand.

Reply:
Move this code to the Validate method.
Also, at the same params to the Validate method, so that you never use global variables and be able to reuse the method